lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -2308,21 +2282,33 @@ LogicalResult StrictConnectOp::verify() {
   if (failed(checkConnectFlow(*this)))
     return failure();
 
-  // Check constraints on RefType.
-  if (failed(checkRefTypeFlow(*this)))
-    return failure();
-
   return success();
 }
 
-LogicalResult RefConnectOp::verify() {
+LogicalResult RefDefineOp::verify() {
   // Check that the flows make sense.
   if (failed(checkConnectFlow(*this)))
     return failure();
 
-  // Check constraints on RefType.
-  if (failed(checkRefTypeFlow(*this)))
-    return failure();
+  // For now, refs can't be in bundles so this is sufficient.
+  // In the future need to ensure no other define's to same "fieldSource".
+  // (When aggregates can have references, we can define a reference within,
+  // but this must be unique.  Checking this here may be expensive,
```

Comment:
I've wondered if a number of the more expensive "must be true" but "should be made true and then never changed" things should move to an optional verif pass.  Similar to "connect doesn't exist anymore" and maybe "only strict registers exist now".  Essentially phase-dependent verification.